src/api.ts

```diff
@@ -50,53 +51,31 @@ export async function getTeamId(teamName: string, leagueCode: string): Promise<n
   const teams = await getCompetitionTeams(leagueCode)
   const team = teams.find((t) => t.name.toLowerCase().includes(teamName.toLowerCase().trim()))
   if (team == null) {
-    throw new Error('Team not found.')
+    throw new ApplicationError(ErrorCode.TEAM_NOT_FOUND, teamName)
   }
   return team.id
 }
 
 async function callApi(url: string, placeholder: string, expiry: number): Promise<any> {
-  if (!cfg.authToken) {
-    apiKeyError()
+  const spinner = createSpinner(placeholder).start()
+  if (!cfg.authToken && !process.env.CI) {
+    spinner.error()
+    throw new ApplicationError(ErrorCode.API_KEY_MISSING)
   }
 
-  const spinner = createSpinner(placeholder).start()
   try {
     const response = await cachedApiCall(url, cfg.authToken, expiry)
     spinner.success().clear()
     return response
-  } catch (error) {
+  } catch (error: unknown) {
     spinner.error()
-    handleError(error)
-  }
-}
-
-function handleError(error: any): void {
-  if (error.response) {
-    console.log(error.response.data.error)
-    console.log(error.response.status)
-  } else if (error.request) {
-    console.log(error.request)
-  } else if (error.message) {
-    console.log('Error', error.message)
-  } else {
-    console.log(error)
+    if (
+      isErrorNodeSystemError(error) &&
+      error.code === 'ENOTFOUND' &&
+      error.syscall === 'getaddrinfo'
```

Comment:
This might be a bit too specific, isn't getaddrinfo used only for DNS errors?

Reply:
It is for that, but Node.js seems to throw that when you don't have an active internet connection. I guess it's probably because it cannot reach a DNS to resolve `api.football-data.org`.

You can test this by disconnecting yourself before trying to use the app.